File: App/app/src/main/java/com/prm392g2/prmapp/dtos/decks/DeckEditDTO.java
package com.prm392g2.prmapp.dtos.decks;

import com.prm392g2.prmapp.dtos.cards.CardDetailDTO;

import java.util.List;

public class DeckEditDTO
{
    public String name;
    public String description;
    public boolean isPublic;
    public List<CardDetailDTO> cards;
}

File: Server/PRMServer.Application/DTOs/Decks/DeckEditDTO.cs
using System.ComponentModel.DataAnnotations;
using PRMServer.Application.DTOs.Cards;
using PRMServer.Data.Models;

namespace PRMServer.Application.DTOs.Decks
{
    public class DeckEditDTO
    {
        [MaxLength(Deck.NAME_MAX_LENGTH)]
        public required string Name { get; set; }

        [MaxLength(Deck.NAME_MAX_LENGTH)]
        public required string Description { get; set; }

        public required bool IsPublic { get; set; }

        public required IList<CardDetailDTO> Cards { get; set; }
    }
}
